agentspawner/daemon.py

```diff
@@ -0,0 +1,126 @@
+#! /bin/python3
+
+"""
+Manage ideal number of "agent" like resources in Resalloc.
+"""
+
+import subprocess
+import time
+import logging
+
+from resalloc.client import (
+    Connection as ResallocConnection,
+    Ticket,
+)
+
+RESALLOC_SERVER = "http://localhost:49100"
+
+
+class SpawnerPool:
+    """
+    Manage ideal number of "agent" like resources in Resalloc.
+    """
+    sleep = 30
+
+    def __init__(self, resalloc_connection, logger):
+        self.tags = ["A"]
+        # TODO: use a persistent storage so we can restart the process
+        self.tickets = []
+        self.conn = resalloc_connection
+        self.log = logger
+
+    def call_converge_to(self):
+        """ Execute the configured hook script """
+        while True:
+            result = subprocess.run(["./hook-converge-to"], capture_output=True,
```

Comment:
I had to replace `capture_output=True` with `stdout=subprocess.PIPE` to get this script working on a RHEL 8 machine.

Reply:
Ah, Python 3.7+, thank you for the hint.